### DISCUSSION TITLE
Custom SASS $theme-colors in Bootstrap 5.0

### DISCUSSION THREAD

**Discussion by jonaslingren** [twbs/bootstrap#33025] [Q&A] [answered]
I have problems with using custom theme colors in Bootstrap 5.0-beta1 (should be the newest) SASS.

I add a color to the called "redred" (just for the example) to the $theme-colors in SASS, and it gives me an error when i use it in my other SASS files. The strange thing is that I can use the normal colors $primary and they are the new changed colors i added to the $theme-colors so I know  that it loads it correctly. Also I can use classes like "btn-redred" and "bg-redred" with no problem in the HTML which gives the correct color, but for some reason I can't use those new added custom colors in the SASS? Am I doing something off? Do you need to use som kind of mixin for custom added colors to the $theme-colors? 

I had no problems with this in Bootstrap 4 using theme-color("redred").

This is my $theme-colors
![image](https://user-images.githubusercontent.com/72208908/107354668-bbb5e780-6ace-11eb-9e29-960353c436e3.png)

This is how I can't use it in SASS
![image](https://user-images.githubusercontent.com/72208908/107354811-e7d16880-6ace-11eb-9af6-dd6df18417f1.png)

I get this error:
![image](https://user-images.githubusercontent.com/72208908/107355484-cd4bbf00-6acf-11eb-999c-141d602a3391.png)
![image](https://user-images.githubusercontent.com/72208908/107355498-d2a90980-6acf-11eb-9583-69edc38eedab.png)

If I use any color names predefined in the $theme-colors although I changed the value to another color, like if i changed $primary to red, It will use that new color so I know that it reads the theme-color before compiling. It simply just won't compile new custom colors added.

**Comment by orestes-garcia-martinez** [comment]
Hi @jonaslingren,
In order to add a custom color to your palette you will need to create a new Sass map with your custom values. e.g.

// Create your own map
$custom-colors: (
  "redred": red;
);

// Merge the maps
$theme-colors: map-merge($theme-colors, $custom-colors);

See docs: https://getbootstrap.com/docs/5.0/customize/sass/

**Reply by jonaslingren** [reply]
That was the first thing I tried, but it gives the same error. It can't find the variable $redred. So I tried just to add it directly to $theme-colors, didn't work either. 

Does it compile fine for you? And you can use custom colors in SASS like: `background: $mycustomcolor;` without problems?

**Comment by jonaslingren** [comment, marked as answer]
Finally got it working - if any have the same problem then this is the solution I found. 

First you have to load in your SASS modifier file in after the required Bootstrap files as said in the documentation. You could always load these first then the full bootstrap: 
![image](https://user-images.githubusercontent.com/72208908/107418611-2723a780-6b17-11eb-84ce-648aac70774a.png)

Then you have to set your colors like this:
![image](https://user-images.githubusercontent.com/72208908/107418471-fb082680-6b16-11eb-8d84-fa96bf10b25c.png)

I don't think the bootstrap documentation is totally clear on this.

**Reply by RobinBastiaan** [reply]
Forgive me for asking as I'm new too. So does that mean that now you can use `$mycustomcolor` in your stylesheet, or that you can use for `bg-mycustomcolor` in your templates?

**Reply by SupremeERG** [reply]
> Forgive me for asking as I'm new too. So does that mean that now you can use `$mycustomcolor` in your stylesheet, or that you can use for `bg-mycustomcolor` in your templates?

Having the same problem, did you ever get the answer?

**Reply by SupremeERG** [reply]
> Finally got it working - if any have the same problem then this is the solution I found.
> 
> First you have to load in your SASS modifier file in after the required Bootstrap files as said in the documentation. You could always load these first then the full bootstrap:
> ![image](https://user-images.githubusercontent.com/72208908/107418611-2723a780-6b17-11eb-84ce-648aac70774a.png)
> 
> Then you have to set your colors like this:
> ![image](https://user-images.githubusercontent.com/72208908/107418471-fb082680-6b16-11eb-8d84-fa96bf10b25c.png)
> 
> I don't think the bootstrap documentation is totally clear on this.

Whats in '_settings.scss'

**Reply by RobinBastiaan** [reply]
Thanks!

**Comment by XhmikosR** [comment]
Feel free to submit any PRs which improve the docs.

**Reply by jonaslingren** [reply]
I'm pretty new to the whole github thing, how do you do that for the documentation? Only thought that was for the Bootstrap 5 base code

**Reply by mdo** [reply]
It's all in this same repo, @jonaslingren :).
